.github/workflows/ci.yml

```diff
@@ -44,28 +71,66 @@ jobs:
             libuv1-dev \
             libwebp-dev \
             libz-dev
-          # OpenGL
-          sudo apt-get install -y libopengl0 libx11-dev
-          # Vulkan
-          sudo apt-get install -y mesa-vulkan-drivers glslang-dev
+          if [ "${{ matrix.backend }}" = "opengl" ]; then
+            sudo apt-get install -y libopengl0 libx11-dev
+          fi
+          if [ "${{ matrix.backend }}" = "vulkan" ]; then
+            sudo apt-get install -y mesa-vulkan-drivers glslang-dev
+          fi
+      - name: Install macOS dependencies
+        if: startsWith(matrix.runs-on, 'macos')
+        run: |
+          # Install dependencies via Homebrew
+          brew install \
+            curl \
+            glfw \
+            jpeg \
+            libpng \
+            sqlite \
+            libuv \
+            webp \
+            zlib
+          # For Vulkan on macOS, install MoltenVK
+          if [ "${{ matrix.backend }}" = "vulkan" ]; then
+            brew install molten-vk vulkan-headers
+          fi
+      # Regular tests
+      - run: just ci-test ${{ matrix.backend }}
+      - run: just test-doc ${{ matrix.backend }}
+      # Test MSRV
+      - name: Read MSRV
+        id: msrv
+        run: echo "value=$(sed -nE 's/^rust-version[[:space:]]*=[[:space:]]*"([^"]+)".*/\1/p' Cargo.toml)" >> $GITHUB_OUTPUT
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: ${{ steps.msrv.outputs.value }}
+      - run: just ci-test ${{ matrix.backend }}
+      # Test MIRI
+      # Currently disabled due to "unsupported operation: can't call foreign function `mln$bridge$cxxbridge1$MapRenderer_new` on OS `linux`"
+      # Needs reinvestigation if this can be fixed via "-Zmiri-native-lib" or maybe "-Zmiri-native-lib-enable-tracing"
```

Comment:
```suggestion
      # Needs reinvestigation if this can be fixed via "-Zmiri-native-lib" or maybe "-Zmiri-native-lib-enable-tracing"
      # Tracking issue upstream https://github.com/rust-lang/miri/issues/11
```